engine/flink/components/base/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/aggregates.scala

```diff
@@ -128,9 +127,9 @@ object aggregates {
     override def isNeutralForAccumulator(element: FirstAggregator.Element, currentAggregate: Option[AnyRef]): Boolean =
       currentAggregate.isDefined
 
-    override def addElement(el: Element, agg: Aggregate): Aggregate = if (agg.isEmpty) Some(el) else agg
+    override def addElement(el: Element, agg: Aggregate): Aggregate = agg.orElse(Some(el))
 
-    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = agg1
+    override def mergeAggregates(agg1: Aggregate, agg2: Aggregate): Aggregate = agg1.orElse(agg2)
```

Comment:
Am I right that it will be always equal to `agg1`? Because `addElement` is invoked immediately after `zero`? We can leave it as it is - just asking.